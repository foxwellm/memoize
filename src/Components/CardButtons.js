import React, { Component } from 'react';

export default class CardButtons extends Component {

  render() {
    const { currentPage, setFavorite, favorites, name } = this.props
    const cssClasses = [
      "favorite-btn",
      favorites.includes(name) ? "favorite-selected" : null
    ];

    return (
      <div className="card-btns-container">
        <div className={cssClasses.join(' ')}
          onClick={setFavorite}
          data-type={currentPage}
          data-method={name} >
          <i className="fas fa-star"></i>
        </div>
      </div>
    )
  }
}