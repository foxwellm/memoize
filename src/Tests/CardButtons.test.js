import React from 'react';
import CardButtons from '../Components/CardButtons';
import { shallow } from 'enzyme';

describe('App', () => {
  let wrapper;
  const currentPage = 'string'
  const favorites = ["concat"]
  const setFavoriteMock = jest.fn();
  const name = 'concat'

  beforeEach(() => {

    wrapper = shallow(
      <CardButtons
        setFavorite={setFavoriteMock}
        currentPage={currentPage}
        favorites={favorites}
        name={name}
      />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

})