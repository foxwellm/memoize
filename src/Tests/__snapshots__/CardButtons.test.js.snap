// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardButtons
    currentPage="string"
    favorites={
      Array [
        "concat",
      ]
    }
    name="concat"
    setFavorite={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="favorite-btn favorite-selected"
        data-method="concat"
        data-type="string"
        onClick={[MockFunction]}
      >
        <i
          className="fas fa-star"
        />
      </div>,
      "className": "card-btns-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <i
          className="fas fa-star"
        />,
        "className": "favorite-btn favorite-selected",
        "data-method": "concat",
        "data-type": "string",
        "onClick": [MockFunction],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fas fa-star",
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="favorite-btn favorite-selected"
          data-method="concat"
          data-type="string"
          onClick={[MockFunction]}
        >
          <i
            className="fas fa-star"
          />
        </div>,
        "className": "card-btns-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <i
            className="fas fa-star"
          />,
          "className": "favorite-btn favorite-selected",
          "data-method": "concat",
          "data-type": "string",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fas fa-star",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
